CMakeLists.txt

```diff
@@ -406,6 +406,10 @@ option(SWIFT_ENABLE_EXPERIMENTAL_CONCURRENCY
   "Enable experimental Swift concurrency model"
   FALSE)
 
+option(SWIFT_RUNTIME_MACHO_NO_DYLD
+  "Build stdlib assuming the runtime environment does not support dynamic modules."
```

Comment:
Would be good to state in the help text that this is only for Darwin. (It's clear enough from the MACHO in the name, but this would help make it clear that this is the intent.) </bikeshed>

Reply:
Hm. My use case (for the 'freestanding' stdlib mode) is actually for a non-Darwin OS (because 'freestanding' should really be independent of the OS), but with Mach-O files. Or did you mean the comment to point out that this only changes behavior that would otherwise kick in on Darwin?

Reply:
Ah right, I was thinking Darwin <-> Mach-O when I wrote that. What I should have said is that the help text should state that this is only for platforms that use Mach-O. Something like, "Build stdlib assuming the runtime environment uses Mach-O but does not support dynamic modules."